### PR TITLE
xtest storage: add case 6019 for storage independence

### PR DESCRIPTION
**Pull request by airbak**
The TA's storage should be independent of each other. What ever
TA #2 do should not affect TA #1's stroage.

Signed-off-by: Jianhui Li <airbak.li@hisilicon.com>

### CODE REVIEW

**Comment by airbak** [comment]
@jenswi-linaro 
https://github.com/OP-TEE/optee_os/issues/1780

**Comment by airbak** [comment]
Currently, it will failed with "optee_test/host/xtest/regression_6000.c:1834: fs_open(&sess, file_01, sizeof(file_01), 0x00000004, &obj, storage_id) has an unexpected value: 0xffff0008 = TEEC_ERROR_ITEM_NOT_FOUND, expected 0x0 = TEEC_SUCCESS".

If I used zeroes uuid in encrypt_block() and decrypt_block() , this case could pass.

**Comment by jbech-linaro** [comment]
> Currently, it will failed with ...

This seems OK on HiKey 620.

**Comment by airbak** [comment]
Additional, this case is not OK with RPMB.

**Comment by airbak** [comment]
Remove TEE_DATA_FLAG_ACCESS_READ and add TEE_DATA_FLAG_OVERWRITE flag when create file.

**Comment by etienne-lms** [comment]
Do you mean the test passes if one adds `TEE_DATA_FLAG_ACCESS_READ` to (at least) the first object creation ?

You will need the `TEE_DATA_FLAG_OVERWRITE` only if you expect this test to pass on a corrupted/non-reset secure storage (so that a trailing object does not conflict).

**Comment by airbak** [comment]
1)This test case does not need TEE_DATA_FLAG_ACCESS_READ flag, so I remove it.
2)Ctrl +C, power down or other abnormal during xtest running,  the storage file is exist or corrupted.  It will affect follow test case. In addition,  RPMB is need to reset without TEE_DATA_FLAG_OVERWRITE.

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
To really check the 2 objects do conflict maybe each session should create an object with different content and close the object and the session and each to reopen session and object and check its content is the expected.

**Comment by airbak** [comment]
@etienne-lms ,
 you are right, I have update it, please review, thanks.

File: host/xtest/regression_6000.c [hunk 1]
```diff
@@ -1815,6 +1815,88 @@ static void xtest_tee_test_6018_single(ADBG_Case_t *c, uint32_t storage_id)
...
+	ADBG_EXPECT_TEEC_SUCCESS(c, fs_unlink(&sess, obj));
+exit2:
+	TEEC_CloseSession(&sess2);
+exit3:
+	TEEC_CloseSession(&sess);
+}
+
+DEFINE_TEST_MULTIPLE_STORAGE_IDS(xtest_tee_test_6019)
+
+
```
Comment (etienne-lms) [inline host/xtest/regression_6000.c]: minor: trailing empty line

**Comment by airbak** [comment]
@etienne-lms

**Comment by airbak** [comment]
@etienne-lms thanks for your review .
Remove empty line and apply tags.

**Comment by etienne-lms** [comment]
thanks @airbak for the contribution.

**Pull request merged**